crypto/sessionkey.go

```diff
@@ -162,37 +185,70 @@ func encryptWithSessionKey(message *PlainMessage, sk *SessionKey, config *packet
 		}
 	}
 
-	encryptWriter, err = packet.SerializeLiteral(
-		encryptWriter,
-		message.IsBinary(),
-		message.Filename,
-		message.Time,
-	)
+	if signEntity != nil { // nolint:nestif
+		hints := &openpgp.FileHints{
+			IsBinary: message.IsBinary(),
+			FileName: message.Filename,
+			ModTime:  message.getFormattedTime(),
+		}
 
-	if err != nil {
-		return nil, errors.Wrap(err, "gopenpgp: unable to serialize")
-	}
+		signWriter, err = openpgp.Sign(encryptWriter, signEntity, hints, config)
```

Comment:
I'm not a super big fan of using the top-level `openpgp.Sign` as a building block, though I guess there's not an obvious better option. It has subtle implications, such as it uses the signer's hash algorithm preferences rather than the recipient's, and if we ever implement intended recipient fingerprints, this would break that. I guess it's fine for now but we should keep this in mind in the future.

Also, since this is apparently the first time we use `openpgp.Sign`, that function has been left behind a bit, e.g. it falls back to RIPEMD [here](https://github.com/ProtonMail/go-crypto/blob/da7cae9ac5f28d6ee1d856d8a079f10c8c015f47/openpgp/write.go#L455) - we should at least fix that to fall back to SHA256, akin to [this](https://github.com/ProtonMail/go-crypto/blob/da7cae9ac5f28d6ee1d856d8a079f10c8c015f47/openpgp/write.go#L349). Also, there's a theoretical case where there's no intersection between the sender's preferred hashes and the candidate hashes, in which case I think we would panic, so we should add an error akin to [this one](https://github.com/ProtonMail/go-crypto/blob/da7cae9ac5f28d6ee1d856d8a079f10c8c015f47/openpgp/write.go#L391-L393). Could you make a PR for that, perhaps?

Other than that, it looks good to me :+1:

Reply:
I see, thanks for the insight. I agree that using `Sign()` directly is not ideal, but I couldn't really find anything better. When we'll have intended recipients I guess we'll need another function (to replace/integrate `Sign` as well).

I'll make a PR for the defaults as you suggested :)